packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/functionMappings/sqlite.ts

```diff
@@ -77,7 +77,37 @@ const sqlite3 = {
       END${colAlias}`
     );
   },
+  DATETIME_DIFF: ({ fn, knex, pt, colAlias }: MapFnArgs) => {
+    const date1 = fn(pt.arguments[0]);
+    const date2 = fn(pt.arguments[1]);
+    const unit = fn(pt.arguments[2]);
 
+    return knex.raw(
+      `WITH const AS (JULIANDAY(${date1}) - JULIANDAY(${date2}) AS daysdiff)
+      
+      CASE
+          WHEN ${unit} LIKE '%day%' or ${unit} LIKE '%dy%'
+               THEN const.daysdiff
+          ELSE ${unit} LIKE '%year%' or ${unit} LIKE '%y%'
+               THEN ROUND(const.daysdiff / 365)
+          ELSE ${unit} LIKE '%quarter%' or ${unit} LIKE '%q%'
+               THEN ROUND(const.daysdiff / 91.25)
+          ELSE ${unit} LIKE '%month%' or ${unit} LIKE '%m%'
+               THEN ROUND(const.daysdiff / 30.417)
+          ELSE ${unit} LIKE '%week%' or ${unit} LIKE '%wk%'
+               THEN ROUND(const.daysdiff / 7)
+          ELSE ${unit} LIKE '%hour%' or ${unit} LIKE '%hh%'
+               THEN ROUND(const.daysdiff * 24)
+          ELSE ${unit} LIKE '%minute%' or ${unit} LIKE '%mi%'
+               THEN ROUND(const.daysdiff * 1440)
+          ELSE ${unit} LIKE '%second%' or ${unit} LIKE '%s%'
+               THEN ROUND(const.daysdiff * 86400)
+          ELSE ${unit} LIKE '%millisecond%' or ${unit} LIKE '%ms%'
+               THEN ROUND(const.daysdiff * 8.64e+7)
+      END${colAlias}
```

Comment:
some values are not intuitive. suggest to add some comments.

Reply:
Good suggestion!